float64 x_parent
float64 y_parent
float64 z_parent
float64 x_child
float64 y_child
float64 z_child